source/core_mqtt.c

```diff
@@ -4317,3 +4847,65 @@ void MQTT_SerializeMQTTVec( uint8_t * pAllocatedMem,
 }
 
 /*-----------------------------------------------------------*/
+
+const char * MQTT_GetPacketTypeString( uint8_t packetType )
+{
+    char * retVal;
+    if( ( packetType & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+    {
+        retVal = "PUBLISH";
+    }
+    else
+    {
+        switch( packetType )
+        {
+            case MQTT_PACKET_TYPE_CONNECT:
```

Comment:
Are these packet types different from the control packet types (section 2.1.2 of v5 spec)? These do not share the same value. OR are we clubbing in the flags as well (which would explain the 4 bit shift on the packet type value?

Our code - https://github.com/FreeRTOS/coreMQTT/blob/08510aa48f73a20a217245459c87ce94348684aa/source/include/core_mqtt_serializer.h#L53-L66
Spec - https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901022